pandas/io/parquet.py

```diff
@@ -106,7 +100,22 @@ def write(
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
         # write_to_dataset does not support a file-like object when
         # a directory path is used, so just pass the path string.
+        if is_fsspec_url(path) and "filesystem" not in kwargs:
+            # TODO: use_legacy_dataset=False will require work, when it is available
+            assert kwargs.get("use_legacy_dataset", False) is False
```

Comment:
```suggestion
        if is_fsspec_url(path) and "filesystem" not in kwargs and kwargs.get("use_legacy_dataset", True):
            # TODO: use_legacy_dataset=False will require work, when it is available
            assert kwargs.get("use_legacy_dataset", False) is False
```

`use_legacy_dataset` should be True for this code branch, as otherwise it cannot (yet) handle fsspec filesystems

Reply:
Ah, actually, it can be removed here, as this is the "write" path. But should be added below in the "read" part (where I put the original comment).